### PR TITLE
Stop using `pos_name` attribute in stfl

### PR DESCRIPTION
**Pull request by dennisschagt**
Fixes https://github.com/newsboat/newsboat/issues/72.
Fixes https://github.com/newsboat/newsboat/issues/1126.

This works around [an stfl bug](https://github.com/dennisschagt/stfl/issues/3) by just not using the (slightly) broken feature.

### CODE REVIEW

File: src/listformatter.cpp [hunk 1]
```diff
@@ -73,19 +71,12 @@ std::string ListFormatter::format_list() const
 {
 	std::string format_cache = "{list";
 	for (const auto& line : lines) {
-		std::string str = line.first;
+		std::string str = line;
```
Comment (Minoru) [inline src/listformatter.cpp:46]: I think this is just a leftover from refactoring, and you can remove it by changing the loop to `for (auto str : lines)`.

Comment (dennisschagt) [reply src/listformatter.cpp]: Done: https://github.com/newsboat/newsboat/pull/1132/commits/74a00c75816e30a35de004e2359c31501f440944

File: src/dirbrowserformaction.cpp [hunk 2]
```diff
@@ -268,8 +269,12 @@ void DirBrowserFormAction::prepare()
 
 		ListFormatter listfmt;
 
+		id_at_position.clear();
 		for (std::string directory : directories) {
-			add_directory(listfmt, directory);
+			const auto id = add_directory(listfmt, directory);
```
[the remark this answers is not in the document]
Comment (dennisschagt) [reply src/dirbrowserformaction.cpp]: Sounds good. Implemented in https://github.com/newsboat/newsboat/pull/1132/commits/54ffe63c8d76938b5cab33ec8977c28068046fa4.

**Comment by dennisschagt** [comment]
Thank you for debugging the SEGFAULT!
Fixed in https://github.com/newsboat/newsboat/pull/1132/commits/da1c8495c6cc4c268cbebc3503f5f001e15f6278.

> I'm not sure how "21845" got generated — get_position should return 0 or a correct position — but even the default of 0 would be a wrong index, since the vector is empty. I'm sure you can take it from here.

`pos` is not being written to because the SEGFAULT occurs before a value is assigned to it. (`visible_feeds.size() == 0` so getting the element at position 0 is not allowed).
```
	unsigned int pos = visible_feeds[list.get_position()].second;
	const std::string feedpos = std::to_string(pos);
```

`21845 == 0x5555 == 0b0101010101010101` might be used as a magic debug value by your compiler: https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_debug_values

**Pull request merged**